src/interface/src/app/services/jwt-interceptor.service.ts

```diff
@@ -0,0 +1,40 @@
+import { Injectable } from '@angular/core';
+import {
+  HttpErrorResponse,
+  HttpEvent,
+  HttpHandler,
+  HttpInterceptor,
+  HttpRequest,
+} from '@angular/common/http';
+import { catchError, Observable, switchMap } from 'rxjs';
+import { AuthService } from './auth.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class JwtInterceptor implements HttpInterceptor {
+  constructor(private auth: AuthService) {}
+  intercept(
+    request: HttpRequest<any>,
+    next: HttpHandler
+  ): Observable<HttpEvent<any>> {
+    return next.handle(request).pipe(
+      catchError((error) => {
+        if (
+          error instanceof HttpErrorResponse &&
+          !request.url.includes('login') &&
+          error.status === 401
+        ) {
+          return this.handle401Error(request, next);
+        }
+        throw error;
+      })
+    );
+  }
+
+  handle401Error(request: HttpRequest<any>, next: HttpHandler) {
+    return this.auth
+      .refreshLoggedInUser()
+      .pipe(switchMap(() => next.handle(request)));
```

Comment:
Do you think we should explicitly catch anything here, or maybe within refreshLoggedInUser?

Reply:
Like, catch any possible error? I think the idea is to only catch those 401s. Not sure I got this.
We could catch all errors here and show a snackbar for example, but that usually gets very cumbersome when you do want to treat those errors differently

Reply:
ok, fair point.